mesonbuild/linkers/linkers.py

```diff
@@ -127,19 +128,15 @@ class DynamicLinker(metaclass=abc.ABCMeta):
     def id(self) -> str:
         pass
 
-    def _apply_prefix(self, arg: T.Union[str, T.List[str]]) -> T.List[str]:
-        args = [arg] if isinstance(arg, str) else arg
-        if self.prefix_arg is None:
-            return args
-        elif isinstance(self.prefix_arg, str):
-            return [self.prefix_arg + arg for arg in args]
-        ret: T.List[str] = []
-        for arg in args:
-            ret += self.prefix_arg + [arg]
-        return ret
+    def _apply_prefix(self, *args: T.Union[str, T.List[str]]) -> T.List[str]:
+        fmt = (lambda x: x) if self.prefix_arg is None else self.prefix_arg.wrap
```

Comment:
We "just" want to return args flattened if None, so perhaps `return mesonlib.stringlistify(args)` is more appropriate? Hmm. It is 3 functions deep to `listify(args, flatten=True)` but with strong typing guarantees.